### PR TITLE
Implement account provider

### PR DESCRIPTION
**Pull request by sanlee42**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

**Comment by sanlee42** [comment]
the private key provider can implement by local db provider through import key.

File: account/provider/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "starcoin-account-provider"
+version = "0.1.0"
```
Comment (jolestar) [inline account/provider/Cargo.toml:3]: 可以改成统一的版本号

File: cmd/starcoin/src/cli_state.rs [hunk 2]
```diff
@@ -48,6 +49,7 @@ impl CliState {
         client: Arc<RpcClient>,
         watch_timeout: Option<Duration>,
         node_handle: Option<NodeHandle>,
+        rpc_client: Box<dyn AccountProvider>,
```
Comment (jolestar) [inline cmd/starcoin/src/cli_state.rs:52]: 也叫 account_client 应该更容易理解写

**Review by jolestar — APPROVED** [review]
LGTM

1. cmd 的集成测试现在临时被注释掉了，可以在本地运行一下，确保命令行兼容没问题。
2. 进一步需要增加一些 local provider 的测试用例。
3. 需要补充一些 local provider 的使用文档（Readme.md 里的例子也可以更新一下）。

File: account/provider/Cargo.toml (hunk 1 above)
Comment (sanlee42) [reply account/provider/Cargo.toml]: Updated docs for it: https://github.com/starcoinorg/starcoin-org-v2/commit/733ff1d2ce66ea44aab9c3f5b546c2771ee50440

**Pull request merged**